test/functional/feature_bind_extra.py

```diff
@@ -37,37 +37,41 @@ def setup_network(self):
         loopback_ipv4 = addr_to_hex("127.0.0.1")
 
         # Start custom ports by reusing unused p2p ports
-        port = p2p_port(self.num_nodes)
+        def extra_port():
+            port = p2p_port(extra_port.index)
+            extra_port.index += 1
+            return port
+        extra_port.index = self.num_nodes
 
         # Array of tuples [command line arguments, expected bind addresses].
         self.expected = []
 
         # Node0, no normal -bind=... with -bind=...=onion, thus only the tor target.
+        port = extra_port()
```

Comment:
I find these reassignments confusing - especially if it contains two ports now.
I think we can get rid of that by using lambdas here, which can provide two separate ports as inputs and the loop would generate the new ports

Reply:
We're introducing extra confusion here now with assigning different types to the same variable.
I'm not suggesting a "rewrite", just to make it better than it was before - which it mostly is, hence the concept ack from me.